File: frontend/smriti_app/lib/core/sync/sync_status.dart
// SMRITI - Sync Status States
// Owner: Aradhya (AI + Database + Sync)

enum SyncStatusState {
  pending,
  syncing,
  synced,
  failed,
}

class SyncProgress {
  final SyncStatusState state;
  final int pendingCount;
  final DateTime? lastSyncedAt;
  final String? errorMessage;

  const SyncProgress({
    required this.state,
    this.pendingCount = 0,
    this.lastSyncedAt,
    this.errorMessage,
  });

  SyncProgress copyWith({
    SyncStatusState? state,
    int? pendingCount,
    DateTime? lastSyncedAt,
    String? errorMessage,
  }) {
    return SyncProgress(
      state: state ?? this.state,
      pendingCount: pendingCount ?? this.pendingCount,
      lastSyncedAt: lastSyncedAt ?? this.lastSyncedAt,
      errorMessage: errorMessage ?? this.errorMessage,
    );
  }
}

File: backend/app/sync/sync_status.py
# SMRITI - Sync Status Constants & ACK Model
# Owner: Aradhya (AI + Database + Sync)

from enum import Enum
from typing import List, Any, Optional
from pydantic import BaseModel

class SyncStatus(str, Enum):
    PENDING = "PENDING"
    SYNCING = "SYNCING"
    SYNCED = "SYNCED"
    FAILED = "FAILED"

class SyncAckResponse(BaseModel):
    status: str
    synced_count: int
    synced_event_ids: List[str]
    synced_local_ids: List[Any]
    message: str
